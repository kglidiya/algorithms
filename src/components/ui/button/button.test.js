import renderer from "react-test-renderer";
import { render, screen, fireEvent } from '@testing-library/react';

import { Button } from "./button";

describe('Button component tests', () => {
  it("button without text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("button with text", () => {
    const tree = renderer.create(<Button text="text" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("button disabled", () => {
    const tree = renderer.create(<Button disabled />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("button isLoader", () => {
    const tree = renderer.create(<Button isLoader />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('button with callback', () => {
    window.alert = jest.fn()
    render(<Button text='test button' onClick={alert('test')} />)
    const button = screen.getByText('test button')
    fireEvent.click(button)
    expect(window.alert).toHaveBeenCalledWith('test')
  })
})
