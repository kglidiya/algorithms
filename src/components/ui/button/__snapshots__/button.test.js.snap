// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component tests button disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button component tests button isLoader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button component tests button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    text
  </p>
</button>
`;

exports[`Button component tests button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;
